src/domdiv/main.py

```diff
@@ -281,6 +281,11 @@ def filter_sort_cards(cards: list[Card], options) -> list[Card]:
             if Card.sets[card.cardset_tag]["upgrades"]:
                 options.exclude_expansions.append(card.cardset_tag.lower())
                 card.cardset_tag = Card.sets[card.cardset_tag]["upgrades"]
+    if options.removed_with_expansion:
+        for card in cards:
+            if Card.sets[card.cardset_tag].get("removed", False):
+                options.exclude_expansions.append(card.cardset_tag.lower())
```

Comment:
This will conflict with the change from list to set in #565. Either this PR or #565 can be merged first, but then the other one needs a minor update before it can merge.

Reply:
```suggestion
                options.exclude_expansions.add(card.cardset_tag.lower())
```
If #565 is merged first, this is the necessary change.